packages/component-base/test/url-utils.test.js

```diff
@@ -35,20 +45,68 @@ describe('url-utils', () => {
       });
     });
 
-    describe('base url', () => {
-      let baseUri;
+    it('should use document.baseURI as a base url', () => {
+      documentBaseURI.value('https://vaadin.com/docs/');
+      expect(matchPaths('https://vaadin.com/docs/components', 'components')).to.be.true;
+    });
 
-      beforeEach(() => {
-        baseUri = sinon.stub(document, 'baseURI');
-      });
+    describe('query params', () => {
+      it('should return true when query params match', () => {
+        expect(matchPaths('/products', '/products')).to.be.true;
+        expect(matchPaths('/products?c=socks', '/products')).to.be.true;
+        expect(matchPaths('/products?c=pants', '/products')).to.be.true;
+        expect(matchPaths('/products?c=', '/products')).to.be.true;
+        expect(matchPaths('/products?c=socks&item=5', '/products')).to.be.true;
+        expect(matchPaths('/products?item=5&c=socks', '/products')).to.be.true;
+        expect(matchPaths('/products?c=socks&c=pants', '/products')).to.be.true;
+        expect(matchPaths('/products?socks', '/products')).to.be.true;
+        expect(matchPaths('/products?socks=', '/products')).to.be.true;
 
-      afterEach(() => {
-        baseUri.restore();
+        expect(matchPaths('/products?c=socks', '/products?c=socks')).to.be.true;
+        expect(matchPaths('/products?c=socks&item=5', '/products?c=socks')).to.be.true;
+        expect(matchPaths('/products?item=5&c=socks', '/products?c=socks')).to.be.true;
+        expect(matchPaths('/products?c=socks&c=pants', '/products?c=socks')).to.be.true;
```

Comment:
**note:** The browser URL `/products?c=socks&c=pants` matches the item path `/products?c=socks` in my implementation. This case was marked with a question mark in @sissbruecker's table. In my opinion, this behavior is consistent with other cases.